src/zmalloc.c

```diff
@@ -626,9 +626,49 @@ size_t zmalloc_get_rss(void) {
 
 #if defined(USE_JEMALLOC)
 
-int zmalloc_get_allocator_info(size_t *allocated,
-                               size_t *active,
-                               size_t *resident) {
+#include <assert.h>
+
+#define STRINGIFY_(x) #x
+#define STRINGIFY(x) STRINGIFY_(x)
+
+/* Compute the total memory wasted in fragmentation of inside small arena bins.
+ * Done by summing the memory in unused regs in all slabs of all small bins. */
+size_t zmalloc_get_frag_smallbins(void) {
+    unsigned nbins;
+    size_t sz, frag = 0;
+    char buf[100];
+
+    sz = sizeof(unsigned);
+    assert(!je_mallctl("arenas.nbins", &nbins, &sz, NULL, 0));
+    for (unsigned j = 0; j < nbins; j++) {
+        size_t curregs, curslabs, reg_size;
+        uint32_t nregs;
+
+        snprintf(buf,sizeof(buf),"arenas.bin.%d.size", j);
+        sz = sizeof(size_t);
+        assert(!je_mallctl(buf, &reg_size, &sz, NULL, 0));
+
+        snprintf(buf,sizeof(buf),"stats.arenas." STRINGIFY(MALLCTL_ARENAS_ALL) ".bins.%d.curregs", j);
+        sz = sizeof(size_t);
+        assert(!je_mallctl(buf, &curregs, &sz, NULL, 0));
+
+        snprintf(buf,sizeof(buf),"arenas.bin.%d.nregs", j);
+        sz = sizeof(uint32_t);
+        assert(!je_mallctl(buf, &nregs, &sz, NULL, 0));
+
+        snprintf(buf,sizeof(buf),"stats.arenas." STRINGIFY(MALLCTL_ARENAS_ALL) ".bins.%d.curslabs", j);
+        sz = sizeof(size_t);
+        assert(!je_mallctl(buf, &curslabs, &sz, NULL, 0));
+
+        frag += ((nregs * curslabs) - curregs) * reg_size;
```

Comment:
malloc-stats shows
```
                            allocated         nmalloc   (#/sec)         ndalloc   (#/sec)       nrequests   (#/sec)           nfill   (#/sec)          nflush   (#/sec)
small:                         582944           15426      3085               0         0           15625      3125             168        33               0         0
large:                         741376              10         2               0         0              10         2              10         2               0         0
total:                        1324320           15436      3087               0         0           15635      3127             178        35               0         0
```
so i was hoping it keeps track of the total allocated in all small bins, and if it doesn't i was hoping it also keeps a total of active, and we don't have to do this loop on our own.
please look into that.

Reply:
we can get the total allocated of small bins with `stats.arenas.<i>.small.allcated`, but here we wanna get the total framentation bytes of small bins, the total allocated of small bins doesn't satisfy the need.